cmd/parca-agent/main.go

```diff
@@ -267,6 +326,87 @@ func main() {
 		"default_cpu_sampling_frequency": strconv.Itoa(defaultCPUSamplingFrequency),
 	})
 
+	if !flags.Telemetry.DisablePanicTelemetry && len(flags.RemoteStore.Address) > 0 {
+		// Spawn ourselves in a child process but disabling telemetry in it.
+		argsCopy := make([]string, 0, len(os.Args)+1)
+		argsCopy = append(argsCopy, os.Args...)
+		argsCopy = append(argsCopy, "--telemetry-disable-panic-telemetry")
+
+		buf, _ := circbuf.NewBuffer(4096)
+
+		cmd := exec.Command(argsCopy[0], argsCopy[1:]...) //nolint:gosec
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = io.MultiWriter(os.Stderr, buf)
+
+		// Run garbage collector to minimize the amount of memory that the parent
+		// telemetry process uses.
+		runtime.GC()
+		err := cmd.Run()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "======================= unexpected error =======================")
```

Comment:
Want to keep things as bare-bones as possible in the error handler as anything could have happened, including out-of-memory conditions. We do a bunch of allocations below in the RPC itself anyways, but still 😄

Reply:
I think we should use our structured logging facilities. I would prefer to be able query these on our logging infrastructure for example.